src/irmin-pack/irmin_pack.ml

```diff
@@ -548,6 +555,53 @@ struct
           Commit.CA.clear_lru (snd (commit_t t))
         in
         Contents.CA.sync ~clear_lrus (contents_t t)
+
+      let migrate v1 v2 =
+        Log.debug (fun l -> l "migrate");
+        let contents = contents_t v1 in
+        let nodes = node_t v1 |> snd in
+        let commits = commit_t v1 |> snd in
+        let f (k, (offset, length, m)) =
+          match m with
+          | 'B' ->
+              contents_t v2
+              |> Contents.CA.migrate_to_current_version ~offset ~length k
+                   contents
+          | 'N' | 'I' ->
+              node_t v2
+              |> snd
+              |> Node.CA.migrate_to_current_version ~offset ~length k nodes
+          | 'C' ->
+              commit_t v2
+              |> snd
+              |> Commit.CA.migrate_to_current_version ~offset ~length k commits
+          | _ -> invalid_arg "unknown content type"
+        in
+        Index.iter (fun k v -> f (k, v)) v1.index
+
+      let migrate_to_current_version config v1 =
+        (*open a fresh store in the new format*)
+        let root = root config in
+        let root_v2 = Filename.concat root "../_v2" in
```

Comment:
I think it's probably not OK to assume we can write a file here (outside of the scope of the directory that was originally given). Perhaps use a `/tmp` file instead? (something like `<tmp_dir>/irmin-migrate-%06x`?)

Reply:
Good point, thanks. Done [here](https://github.com/mirage/irmin/pull/1015/files#diff-8a044b6b9c9b0a69a5f770d759c9d1b9R301).